crates/rpc/rpc/src/eth/api/optimism.rs

```diff
@@ -28,3 +30,16 @@ impl OptimismTxMeta {
         Self { l1_block_info, l1_fee, l1_data_gas }
     }
 }
+
+/// Optimism Sequencer Client
+#[derive(Debug, Default, Clone)]
+pub struct SequencerClient {
+    pub sequencer_endpoint: Option<String>,
+    pub http_client: Option<Client>,
```

Comment:
having both options makes this very hard to work with.

this should not have options, instead it should be used as Option<SequencerCLient>